src/NeoSharp.Core/Network/Rpc/Mvc/MvcRpcRedirectFilter.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Newtonsoft.Json.Linq;
+
+namespace NeoSharp.Core.Network.Rpc.Mvc
+{
+    public class MvcRpcRedirectFilter : IActionFilter
+    {
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var ctx = context.HttpContext;
+            var isPost = HttpMethods.Post.Equals(ctx.Request.Method, StringComparison.OrdinalIgnoreCase);
+
+            string postBody = null;
+
+            if (isPost)
+            {
+                var bodyStream = new StreamReader(ctx.Request.Body);
```

Comment:
Max post size here too

Reply:
@melanke Please fix this before merge

Reply:
Done

Reply:
I tested and I think it's not necessary. The Middleware is executed before the Filter and stops the whole request if the post body is too big, so it doesn't get here.